crates/router/src/connector/adyen/transformers.rs

```diff
@@ -289,139 +352,341 @@ impl TryFrom<&types::BrowserInformation> for AdyenBrowserInfo {
+    let state_or_province = address.and_then(|address| address.state.clone());
+    let street = address.and_then(|address| address.line2.clone());
+    Some(Address {
+        city,
+        country,
+        house_number_or_name,
+        postal_code,
+        state_or_province,
+        street,
+    })
+}
 
-        Ok(Self {
-            amount,
-            merchant_account: auth_type.merchant_account,
-            payment_method,
-            reference,
-            return_url: item.router_return_url.clone().ok_or(
-                errors::ConnectorError::MissingRequiredField {
-                    field_name: "router_return_url",
-                },
-            )?,
-            shopper_interaction,
-            recurring_processing_model,
-            browser_info,
-            additional_data,
+fn get_line_items(item: &types::PaymentsAuthorizeRouterData) -> Option<Vec<LineItem>> {
+    let order_details = item.request.order_details.as_ref();
+    let line_item = LineItem {
+        amount_including_tax: Some(item.request.amount),
+        amount_excluding_tax: None,
+        description: order_details.map(|details| details.product_name.clone()),
+        id: Some(String::from("Items #1")),
+        tax_amount: None,
+        quantity: order_details.map(|details| details.quantity),
+    };
+    Some(vec![line_item])
+}
+
+fn get_telephone_number(item: &types::PaymentsAuthorizeRouterData) -> Option<Secret<String>> {
+    let phone = item
+        .address
+        .billing
+        .as_ref()
+        .and_then(|billing| billing.phone.as_ref());
+    phone.as_ref().and_then(|phone| {
+        phone.number.as_ref().and_then(|number| {
+            phone
+                .country_code
+                .as_ref()
+                .map(|cc| Secret::new(format!("{}{}", cc, number.peek())))
         })
+    })
+}
+
+fn get_shopper_name(item: &types::PaymentsAuthorizeRouterData) -> Option<ShopperName> {
+    let address = item
+        .address
+        .billing
+        .as_ref()
+        .and_then(|billing| billing.address.as_ref());
+    Some(ShopperName {
+        first_name: address.and_then(|address| address.first_name.clone()),
+        last_name: address.and_then(|address| address.last_name.clone()),
+    })
+}
+
+fn get_country_code(item: &types::PaymentsAuthorizeRouterData) -> Option<String> {
+    let address = item
+        .address
+        .billing
+        .as_ref()
+        .and_then(|billing| billing.address.as_ref());
+    address.and_then(|address| address.country.clone())
+}
+
+fn get_payment_method_data(
+    item: &types::PaymentsAuthorizeRouterData,
+) -> Result<AdyenPaymentMethod, error_stack::Report<errors::ConnectorError>> {
+    match item.request.payment_method_data {
+        api::PaymentMethod::Card(ref card) => {
+            let adyen_card = AdyenCard {
+                payment_type: String::from("scheme"),
+                number: Some(card.card_number.clone()),
```

Comment:
Card number should not be an option in `AdyenCard` struct, if card number is missing we must validate it before calling the connector. Please make all the fields inside the payment method specific structs as mandatory